tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -318,19 +318,79 @@ public function adminChangesPasswordOfUserToUsingTheGraphApi(
 	}
 
 	/**
-	 * returns list of all groups
 	 *
-	 * @return array
+	 * @param array $groups
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theseGroupsShouldBeInTheResponse(array $groups): void {
+		$respondedGroups = $this->getArrayOfGroupsResponded($this->featureContext->getResponse());
+		foreach ($groups as $group) {
+			$found = false;
+			foreach ($respondedGroups as $respondedGroup) {
+				if ($respondedGroup["displayName"] === $group) {
+					$found = true;
+					break;
+				}
+			}
+			Assert::assertTrue($found, "Group '$group' not found in the list");
+		}
+	}
+
+	/**
+	 *
+	 * @param array $users
+	 *
+	 * @return void
 	 * @throws Exception
+	 */
+	public function theseUsersShouldBeInTheResponse(array $users): void {
+		$respondedUsers = $this->getArrayOfUsersResponded($this->featureContext->getResponse());
+		foreach ($users as $user) {
+			$found = false;
+			foreach ($respondedUsers as $respondedUser) {
+				if ($respondedUser["onPremisesSamAccountName"] === $user) {
+					$found = true;
+					break;
+				}
+			}
+			Assert::assertTrue($found, "User '$user' not found in the list");
+		}
+	}
+
+	/**
+	 *
+	 * @param string|null $user
+	 *
+	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 */
-	public function adminHasRetrievedGroupListUsingTheGraphApi(): array {
-		$response =  GraphHelper::getGroups(
+	public function listGroups(?string $user = null): ResponseInterface {
+		if ($user) {
+			$username = $user;
+			$password = $this->featureContext->getPasswordForUser($user);
+		} else {
+			$username = $this->featureContext->getAdminUsername();
+			$password = $this->featureContext->getAdminPassword();
+		}
+		return GraphHelper::getGroups(
 			$this->featureContext->getBaseUrl(),
 			$this->featureContext->getStepLineRef(),
-			$this->featureContext->getAdminUsername(),
-			$this->featureContext->getAdminPassword()
+			$username,
+			$password
 		);
```

Comment:
can we reuse this? seems like being used similarly in multiple places.

Reply:
Couldn't think of any other ways. The inner function and params are different for each function so I find it hard to not repeat it. Could you suggest me some ways?

Reply:
I can think sth like this where the condition is wrapeed in a func. and we just call that for the actor information getter func.:
```php
$actorInfo = $this->getActorOrAdminAuthCred($user)
```
& func. returning like:
```php
public function getActorOrAdminAuthCred($user) {
    actor = []
    $actor["username"] = $user ? $user : $this->featureContext->getAdminUsername();
    $actor["password"] = $user ? $this->featureContext->getPasswordForUser($user) : $this->featureContext->getAdminPassword();;
    return $actor
}
```

Reply:
looks cool